storm/src/main/storm/mesos/MesosCommon.java

```diff
@@ -72,8 +72,15 @@ public static String getMesosComponentNameDelimiter(Map conf, TopologyDetails in
         .or(DEFAULT_MESOS_COMPONENT_NAME_DELIMITER);
   }
 
+  public static String timestampMillis() {
+    long now = System.currentTimeMillis();
+    long sec = now / 1000L;
+    long msec = now % 1000L;
```

Comment:
[This](http://stackoverflow.com/questions/625433/how-to-convert-milliseconds-to-x-mins-x-seconds-in-java) approach seems cleaner?